test/jdk/java/net/SocketPermission/SocketPermissionIm.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * @test
+ * @bug 8243376
+ * @summary SocketPermission implies(Permission p) spec allows, If the object was initialized with a single IP address and one of p's IP addresses is equal to this object's IP addr
+ * @run java -Dsun.net.inetaddr.ttl=0 SocketPermissionIm
+ */
+
+import java.net.SocketPermission;
+import java.net.InetAddress;
+import java.io.*;
+import java.io.File;
+
+public class SocketPermissionIm {
+    public static void main(String[] args) throws Exception {
+        String hostname = "www.exmp.com";
+        String hostsFileName = System.getProperty("test.src", ".") + File.separator + "Host.txt";
+        System.setProperty("jdk.net.hosts.file", hostsFileName);
+
+        int testPass = 0;
+        SocketPermission sp = new SocketPermission(hostname, "connect,resolve");
+
+            do {
+                   if (!sp.implies(new SocketPermission(hostname, "connect,resolve"))) {
+                       System.out.println("Expected true, returned false");
+                       break;
+                   }
+                   addIpToHostsFile(hostname, "1.2.3."+testPass, hostsFileName);
+                   Thread.sleep(1000);
+                   testPass++;
+               } while (testPass <= 2);
+    }
+
```

Comment:
Formatting looks off in the file. The do statement in 22 should be aligned with SocketPermission in line 120.

The implementation changes look fine.

Reply:
Corrected the alignment.